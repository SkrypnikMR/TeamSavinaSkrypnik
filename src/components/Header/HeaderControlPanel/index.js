import { withRouter } from 'react-router';
import { connect } from 'react-redux';
import { setValue } from 'src/Store/user/actions';
import { userThemeMode, userNotifSettings } from 'src/Store/user/selectors';
import HeaderControlPanel from './HeaderControlPanel';

const mapStateToProps = state => ({
    themeMode: userThemeMode(state),
    userNotifSettings: userNotifSettings(state),
});
const mapDispatchToProps = dispatch => ({
    setValue: payload => dispatch(setValue(payload)),
});

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(HeaderControlPanel));
