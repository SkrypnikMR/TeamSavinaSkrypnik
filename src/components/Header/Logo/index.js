import { withRouter } from 'react-router';
import { connect } from 'react-redux';
import Logo from './Logo.jsx';
import { setValue } from '../../../Store/user/actions';

const mapDispatchToProps = dispatch => ({ setValue: payload => dispatch(setValue(payload)) });

export default connect(null, mapDispatchToProps)(withRouter(Logo));
