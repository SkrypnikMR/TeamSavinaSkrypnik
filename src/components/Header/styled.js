import styled from 'styled-components';
import { TRANSPARENT_BACKGROUND } from '/src/components/UI/baseLayout';

export const StHeader = styled.div`
    display:flex;
    justify-content: flex-start;
    align-items: center;
    height:10vh;
    ${({ backgroundColor = TRANSPARENT_BACKGROUND }) => backgroundColor && `background-color: ${backgroundColor}`};
    ${({ transition = 'all 300ms ease-in-out' }) => transition && `transition: ${transition}`};
`;
