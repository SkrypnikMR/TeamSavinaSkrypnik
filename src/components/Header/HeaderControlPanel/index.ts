import { withRouter } from 'react-router';
import HeaderControlPanel from './HeaderControlPanel';

export default withRouter(HeaderControlPanel);
