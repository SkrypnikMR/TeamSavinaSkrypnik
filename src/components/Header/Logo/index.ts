import { withRouter } from 'react-router';

import Logo from './Logo';

export default withRouter(Logo);
