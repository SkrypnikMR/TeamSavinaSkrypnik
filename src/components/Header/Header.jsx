import React from 'react';
import { StHeader } from './styled';
import Logo from './Logo';
import HeaderControlPanel from './HeaderControlPanel';

const Header = () => {
    return (
        <StHeader >
            <Logo />
            <HeaderControlPanel />
        </StHeader >
    );
};

export default Header;
