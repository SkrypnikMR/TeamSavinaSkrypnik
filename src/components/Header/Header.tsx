import React from 'react';

import Logo from './Logo';
import HeaderControlPanel from './HeaderControlPanel';
import { useTheme } from '../Hook/useTheme';

import { StHeader } from './styled';

const Header = () => {
    const { colors, theme } = useTheme();
    return (
        <StHeader colors={colors} theme={theme}>
            <Logo />
            <HeaderControlPanel />
        </StHeader >
    );
};

export default Header;
