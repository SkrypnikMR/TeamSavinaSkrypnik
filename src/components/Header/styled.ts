import styled from 'styled-components';
import { IDiv } from './types';

export const StHeader = styled.div < IDiv > `
    display:flex;
    justify-content: flex-start;
    align-items: center;
    height:10vh;
    background: ${({ colors, theme }) => colors[theme].background};
    ${({ transition = 'all 300ms ease-in-out' }) => transition && `transition: ${transition}`}
`;
