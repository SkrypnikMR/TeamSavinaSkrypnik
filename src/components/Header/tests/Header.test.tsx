import React from 'react';
import configureStore from 'redux-mock-store';
import { shallowSmart, mountSmart } from '../../../../__tests__/testHelper';
import Header from '../Header';

const mockStore = configureStore();
const store = mockStore({
   game: {
     rooms: [],
    userLogin: '',
    actualRoom: {
      gameType: '',
      creatorLogin: '',
      guestLogin: '',
      startTime: 0,
      id: '',
      stepDoList: [],
    },
    stepOrder: '',
    stepHistory: [],
    winner: '',
    possibleSteps: [],
  },
});
describe('Header', () => {
    it('Should match snapshot', () => {
        const component = shallowSmart(<Header />, store);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render p', () => {
        const component = mountSmart(<Header />, store);
        expect(component.find('Logo')).toHaveLength(1);
    });
    it('should render inputs', () => {
        const component = mountSmart(<Header />, store);
        expect(component.find('HeaderControlPanel')).toHaveLength(1);
    });
});
