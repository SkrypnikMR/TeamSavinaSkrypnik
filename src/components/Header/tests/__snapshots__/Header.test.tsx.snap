// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should match snapshot 1`] = `"<div class=\\"styled__StHeader-sc-1d7yllk-0 ibTsld\\"><div class=\\"styled__StLogo-sc-1lhyjdk-0 eFYqgc\\"><img src=\\"public/assets/images/game-development.png\\"/><div><button type=\\"button\\" font-size=\\"20px\\" class=\\"styled__StButton-sc-1y35zax-0 kFMylk\\">game_zone</button><button type=\\"button\\" font-size=\\"20px\\" class=\\"styled__StButton-sc-1y35zax-0 qmQvK\\">statistics</button></div></div><div class=\\"styled__StControl-sc-1l7fu2q-0 bcijKV\\"> <button id=\\"theme_btn\\" type=\\"button\\" value=\\"dark\\" color=\\"white\\" font-size=\\"26px\\" width=\\"40px\\" class=\\"styled__StButton-sc-1y35zax-0 jZPxgN\\">ðŸ’¡</button><button id=\\"en\\" type=\\"button\\" value=\\"en\\" color=\\"white\\" font-size=\\"26px\\" width=\\"40px\\" class=\\"styled__StButton-sc-1y35zax-0 dDxuZt\\">english_translate_btn</button><button id=\\"ru\\" type=\\"button\\" value=\\"ru\\" color=\\"white\\" font-size=\\"26px\\" width=\\"40px\\" class=\\"styled__StButton-sc-1y35zax-0 gQNzZW\\">russian_translate_btn</button><img src=\\"public/assets/images/log-out.png\\"/></div></div>"`;
